meerk40t/kernel.py

```diff
@@ -149,6 +149,8 @@ def __repr__(self):
         return "Context('%s')" % self._path
 
     def __call__(self, data: str, **kwargs):
+        if len(data) and data[-1] != "\n":
```

Comment:
`if len(data) == 0 or data[-1] != "\n":`?

Reply:
if len(data) != 0

The truthy value of len(data) is `len(data) != 0` since len() returns an int and bool(int) is int 0 or non-zero

Reply:
OK - I was suggesting that we append a "\n" if **either** the string is null **or** doesn't already end with it. But on reflection this doesn't seem right either and we should perhaps not bother sending a null string to console in any case.